accessibility_monitoring_platform/apps/cases/models.py

```diff
@@ -1193,89 +1177,149 @@ def save(self, *args, **kwargs) -> None:
     def __str__(self) -> str:
         return self.status
 
-    def calculate_and_save_status(self) -> None:
-        self.status = self.calculate_status()
-        self.save()
-
-    def calculate_status(self) -> str:  # noqa: C901
-        try:
-            compliance: CaseCompliance = self.case.compliance
-        except CaseCompliance.DoesNotExist:
-            compliance = None
-
-        if self.case.is_deactivated:
-            return CaseStatus.Status.DEACTIVATED
-        elif (
+    @property
+    def calculate_case_complete_status(self) -> bool:
+        """Calculates case complete status"""
+        return (
             self.case.case_completed == Case.CaseCompleted.COMPLETE_NO_SEND
             or self.case.enforcement_body_pursuing
             == Case.EnforcementBodyPursuing.YES_COMPLETED
             or self.case.enforcement_body_closed_case
             == Case.EnforcementBodyClosedCase.YES
-        ):
-            return CaseStatus.Status.COMPLETE
-        elif (
+        )
+
+    @property
+    def calculate_in_correspondence_with_enforcement_body_status(self) -> bool:
+        """Calculates correspondence with enforcement body status"""
+        return (
             self.case.enforcement_body_pursuing
             == Case.EnforcementBodyPursuing.YES_IN_PROGRESS
             or self.case.enforcement_body_closed_case
             == Case.EnforcementBodyClosedCase.IN_PROGRESS
-        ):
-            return CaseStatus.Status.IN_CORES_WITH_ENFORCEMENT_BODY
-        elif self.case.sent_to_enforcement_body_sent_date is not None:
-            return CaseStatus.Status.CASE_CLOSED_SENT_TO_ENFORCEMENT_BODY
-        elif self.case.case_completed == Case.CaseCompleted.COMPLETE_SEND:
-            return CaseStatus.Status.CASE_CLOSED_WAITING_TO_SEND
-        elif self.case.no_psb_contact == Boolean.YES:
-            return CaseStatus.Status.FINAL_DECISION_DUE
-        elif self.case.auditor is None:
-            return CaseStatus.Status.UNASSIGNED
-        elif (
+        )
+
+    @property
+    def calculate_case_closed_waiting_to_be_sent_status(self) -> bool:
+        """Calculates correspondence with enforcement body status"""
+        return self.case.case_completed == Case.CaseCompleted.COMPLETE_SEND
```

Comment:
By separating the condition from the status value this make the core more difficult to read. Please change these `calculate_` methods to return the value of the status. Then when a Case gets stuck on a specific status we can read through the code to see why.

Reply:
The code has changed but the `check_` functions (e.g. `check_unnassigned_case_status_complete`) still return booleans.

Reply:
This is needed for the **Status workflow** page, where the checks are already being reused.